lib/generator/underscore.rb

```diff
@@ -2,7 +2,7 @@ module Generator
   module Underscore
     refine String do
       def underscore
-        downcase.gsub(/[- ]/, '_').gsub(/[^\w?]/, '')
+        strip.downcase.gsub(/[- ]/, '_').gsub(/[^\w?]/, '')
```

Comment:
This solution works, but I'm not sure this is the right place to be `strip`ping the string.

I would expect `' something '.underscore` to be `'_something_'` (this may be an unreasonable expectation.)

Is there somewhere further up the call stack where the `strip` could happen?

Reply:
Thanks for reviewing my PR

I was thinking about using `strip` in `exercise_case.rb` instead, but I see that `underscore` method also `downcase` the string. I thought `strip` as part of cleaning up before converting to underscore like how `downcase` was

Reply:
I think it would be better in `exercise_case.rb`

Your reasoning about where to put it is fine. Do you have any objection to moving it to  the `ExerciseCase` class?

I would also prefer that the `downcase` wasn't there, but you didn't put that in and it's not related to the current task, so I wasn't suggesting you take it out at this point.